src/App/Pages/Settings/ExportVaultPageViewModel.cs

```diff
@@ -0,0 +1,144 @@
+                new KeyValuePair<string, string>("csv", ".csv")
+            };
+        }
+
+        public async Task InitAsync()
+        {
+            FileFormatSelectedIndex = FileFormatOptions.FindIndex(k => k.Key == "json");
+        }
+
+        public List<KeyValuePair<string, string>> FileFormatOptions { get; set; }
+
+        public int FileFormatSelectedIndex
+        {
+            get => _fileFormatSelectedIndex;
+            set { SetProperty(ref _fileFormatSelectedIndex, value); }
+        }
+
+        public bool ShowPassword
+        {
+            get => _showPassword;
+            set => SetProperty(ref _showPassword, value,
+                additionalPropertyNames: new string[] {nameof(ShowPasswordIcon)});
+        }
+
+        public string MasterPassword
+        {
+            get => _masterPassword;
+            set => SetProperty(ref _masterPassword, value);
+        }
+
+        public Command TogglePasswordCommand { get; }
+
+        public string ShowPasswordIcon => ShowPassword ? "" : "";
+
+        public void TogglePassword()
+        {
+            ShowPassword = !ShowPassword;
+            (Page as ExportVaultPage).MasterPasswordEntry.Focus();
+        }
+
+        public Command ExportVaultCommand { get; }
+
+        public async Task ExportVaultAsync()
+        {
+            if(string.IsNullOrEmpty(_masterPassword))
+            {
+                _platformUtilsService.ShowToast("error", _i18nService.T("AnErrorHasOccurred"),
+                    _i18nService.T("InvalidMasterPassword"));
+                return;
+            }
+
+            var keyHash = await _cryptoService.HashPasswordAsync(_masterPassword, null);
+            MasterPassword = string.Empty;
+            
+            var storedKeyHash = await _cryptoService.GetKeyHashAsync();
+            if(storedKeyHash != null && keyHash != null && storedKeyHash == keyHash)
+            {
+                try
+                {
+                    var data = _exportService.GetExport(FileFormatOptions[FileFormatSelectedIndex].Key);
+                    var fileFormat = FileFormatOptions[FileFormatSelectedIndex].Key;
+                    _defaultFilename = _exportService.GetFileName(null, fileFormat);
+                    _exportResult = Encoding.ASCII.GetBytes(data.Result);
+
+                    if(!_deviceActionService.SaveFile(_exportResult, null, _defaultFilename, null))
+                    {
+                        ClearResult();
+                        await _platformUtilsService.ShowDialogAsync(_i18nService.T("ExportVaultFailure"));
+                    }
+                }
+                catch(Exception ex)
+                {
+                    ClearResult();
+                    System.Diagnostics.Debug.WriteLine(">>> {0}: {1}", ex.GetType(), ex.StackTrace);
+                }
+            }
+            else
+            {
+                _platformUtilsService.ShowToast("error", _i18nService.T("AnErrorHasOccurred"),
```

Comment:
Should probably be `ShowDialogAsync` too

Reply:
`InvalidMasterPassword` is also shown on line 88 - would you prefer that be a dialog as well?

Reply:
Yes